zone/embparser_api.cpp

```diff
@@ -3986,6 +3986,101 @@ int8 Perl__does_augment_fit_slot(EQ::ItemInstance* inst, uint32 augment_id, uint
 	return quest_manager.DoesAugmentFit(inst, augment_id, augment_slot);
 }
 
+perl::array Perl__GetRecipeComponentItemIDs(uint32 recipe_id)
+{
+	perl::array result;
+
+	const auto& l = content_db.GetRecipeComponentItemIDs(RecipeCountType::Component, recipe_id);
+
+	if (!l.empty()) {
+		for (int i = 0; i < l.size(); i++) {
+			result.push_back(l[i]);
+		}
+	}
+
+	return result;
+}
+
+perl::array Perl__GetRecipeContainerItemIDs(uint32 recipe_id)
+{
+	perl::array result;
+
+	const auto& l = content_db.GetRecipeComponentItemIDs(RecipeCountType::Container, recipe_id);
+
```

Comment:
When you know the size of the vector, it's a good idea to reserve the memory up front.

The `perl::array` works like the STL containers

```cpp
result.reserve(results.RowCount());
```

Or

```cpp
result.reserve(l.size());
```